crates/hir-def/src/nameres/mod_resolution.rs

```diff
@@ -154,6 +154,14 @@ impl DirPath {
     fn join_attr(&self, mut attr: &str, relative_to_parent: bool) -> String {
         let base = if relative_to_parent { self.parent().unwrap() } else { &self.0 };
 
+        if cfg!(windows) {
```

Comment:
There shouldn’t be any cfgs in the analysis parts of rust-analyzer.

A rust-analyzer server running on a Unix host should successfully analyzer code of a windows-only project. 

Host operating system shouldn’t be an input to analysis.

If there’s some need to vary semantics of paths, it should be triggered by an explicit property of a Crate data structure stored in salsa database.